callbacks/query.go

```diff
@@ -146,9 +146,13 @@ func BuildQuerySQL(db *gorm.DB) {
 
 					if isRelations {
 						genJoinClause := func(joinType clause.JoinType, parentTableName string, relation *schema.Relationship) clause.Join {
-							tableAliasName := relation.Name
-							if parentTableName != clause.CurrentTable {
-								tableAliasName = utils.NestedRelationName(parentTableName, tableAliasName)
+							tableAliasName := join.Alias
```

Comment:
Nested joins will use the same alias. Can we use a new API for generics? Maybe something like
```diff
type JoinTarget struct {
	Type JoinType
	Association string
	Subquery Expression
	Table string
+ 	NestedJoin JoinTarget
}
```

Reply:
Fixed the same alias issue for nested Joins.